soc/nxp/lpc/Kconfig

```diff
@@ -3,7 +3,7 @@
 
 config SOC_FAMILY_LPC
 	select HAS_SEGGER_RTT if ZEPHYR_SEGGER_MODULE
-	select CLOCK_CONTROL
+	imply CLOCK_CONTROL
```

Comment:
I don't want to force downstream users to switch to clock management with this PR

Enabling `CONFIG_CLOCK_MGMT` requires defining clock states for the peripherals you are using on your board that enable the framework (currently just Flexcomm USART on the LPC55S69). Since defining those states might take some time, I don't want to force users to transition immediately- I'd rather stabilize the framework, then provide documentation for users so they can migrate to the new framework over a release. I believe we did something similar with pin control, where it was initially optional for downstream boards.